rpc_new/provider/smoldot.ts

```diff
@@ -0,0 +1,92 @@
+import { start } from "../../deps/smoldot.ts";
+import { Chain, Client, ClientOptions } from "../../deps/smoldot/client.d.ts";
+import * as msg from "../messages.ts";
+import { nextIdFactory, Provider, ProviderConnection, ProviderListener } from "./base.ts";
+import { ProviderCloseError, ProviderHandlerError, ProviderSendError } from "./errors.ts";
+
+// TODO
+type SmoldotSendErrorData = any;
+type SmoldotHandlerErrorData = any;
+type SmoldotCloseErrorData = any;
+
+let client: undefined | Client;
+const connections = new Map<string, SmoldotProviderConnection>();
+class SmoldotProviderConnection
+  extends ProviderConnection<Chain, SmoldotSendErrorData, SmoldotHandlerErrorData>
+{}
+
+const nextId = nextIdFactory();
+
+export const smoldotProvider: Provider<
+  string,
+  SmoldotSendErrorData,
+  SmoldotHandlerErrorData,
+  SmoldotCloseErrorData
+> = (chainSpec, listener) => {
+  return {
```

Comment:
```suggestion
  const connProm = connection(chainSpec, listener);
  return {
```

Reply:
Initialize the connection before a send is initiated? I suppose the effect system will take care of orchestration... still, not sure if this is preferable. Thoughts @kratico?

Reply:
Generally I'd expect creating a client to initialize a connection. But the main reason I did that is to avoid calling `connection` repeatedly (though it returns the same thing, there's no need for the extra calls)

Reply:
for now, I would leave it inside the `.send`
for recovery, it might be easier to re-initialize the connection on a new `.send` invocation